### PR TITLE
[GEARPUMP-281] Using new version of docker images for integration tests

### PR DESCRIPTION
**Pull request by karol-brejna-i**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the commit message is formatted like:
   `[GEARPUMP-<Jira issue #>] Meaningful description of pull request` 
 - [ ] Make sure tests pass via `sbt clean test`.
 - [ ] Make sure old documentation affected by the pull request has been updated and new documentation added for new functionality. 

Changes include:
* adding docker project to integrationtest/docker (java, hadoop, kafka, gearpump-launcher)
* updating gearpump-launcher to include new version of java
* update the code to use new images

### CODE REVIEW

File: integrationtest/docker/java/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+A minimalistic Oracle JDK 8 container on top of busybox.
```
Comment (manuzhang) [inline integrationtest/docker/java/README.md]: Great investigation and README

File: integrationtest/docker/kafka/LICENSE [hunk 2]
```diff
@@ -0,0 +1,201 @@
+Apache License
```
Comment (manuzhang) [inline integrationtest/docker/kafka/LICENSE]: We don't keep a separate license file for sub-folders. Instead, we add license headers to source files (except README).

File: integrationtest/docker/README.md [hunk 3]
```diff
@@ -1,28 +1,23 @@
...
+For now our tests focus on Kafka 0.8.x, and the project does not support this version. 
+
+Hadoop docker image (https://hub.docker.com/r/sequenceiq/hadoop-docker/) is well maintained. 
+We rely on version 2.6.0 and we feel there is no need to duplicate it.
+
+Gearpump Cluster Launcher  helps developer to setup/test a local Gearpump cluster quickly. 
+The image is based on a minimal JRE8 environment with Python support.
+
+We used to base Gearpump Cluster Launcher on errordeveloper/oracle-jre image but it stuck on version 8u66-b17 and doesn't support tagging.
+We also probably hit a Java bug (http://bugs.java.com/bugdatabase/view_bug.do?bug_id=8133205) that requires never version of JRE.
```
Comment (huafengw) [inline integrationtest/docker/README.md]: typo 'nerver' => 'newer'

File: integrationtest/docker/java/README.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+We used to base Gearpump Cluster Launcher on errordeveloper/oracle-jre image but it stuck on version 8u66-b17 and doesn't support tagging.
+We also probably hit a Java bug (http://bugs.java.com/bugdatabase/view_bug.do?bug_id=8133205) that requires never version of JRE.
```
Comment (huafengw) [inline integrationtest/docker/java/README.md]: also has typo here

File: integrationtest/docker/kafka/README.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+in the same container. This means:
+
+* No dependency on an external Zookeeper host, or linking to another container
+* Zookeeper and Kafka are configured to work together out of the box
+
+Run
+---
+
+```bash
+docker run -p 2181:2181 -p 9092:9092 --env ADVERTISED_HOST=`docker-machine ip \`docker-machine active\`` --env ADVERTISED_PORT=9092 grubykarol/kafka
```
Comment (huafengw) [inline integrationtest/docker/kafka/README.md]: it needs docker-machine?

Comment (karol-brejna-i) [inline integrationtest/docker/kafka/README.md]: No, we use it without it. It's just example from original README.

File: integrationtest/docker/kafka/LICENSE (hunk 2 above)
Comment (karol-brejna-i) [inline integrationtest/docker/kafka/LICENSE]: added licence header to dockerfiles

File: integrationtest/docker/README.md (hunk 3 above)
Comment (karol-brejna-i) [inline integrationtest/docker/README.md]: fixed

File: integrationtest/docker/java/README.md (hunk 4 above)
Comment (karol-brejna-i) [inline integrationtest/docker/java/README.md]: fixed

**Comment by huafengw** [comment]
Hi Karol, would you mind do some code modification in org.apache.gearpump.integrationtest.Docker.scala, change the implementation of `getNetworkGateway` to 
```
  final def getNetworkGateway(container: String): String = {
    trace(container, s"Get gateway of container...") {
      Docker.inspect(container, "--format={{.NetworkSettings.Gateway}}")
    }
  }
```

Then rerun the integration test on your Ubuntu environment to see whether the modification is OK?

**Comment by karol-brejna-i** [comment]
@huafengw no problem. Let me get back to you soon.

**Comment by karol-brejna-i** [comment]
The change you suggested works fine on my ubuntu 14.04:
```
[info] Run completed in 25 minutes, 45 seconds.
[info] Total number of tests run: 66
[info] Suites: completed 9, aborted 0
[info] Tests: succeeded 66, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
```

**Comment by huafengw** [comment]
+1 Merging
Thanks Karol!

**Pull request closed without merging**